js/repl/ASTPanel.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import { css } from "emotion";
+import { colors } from "./styles";
+import React from "react";
+import ReactJson from "react-json-view";
+import {
+  flatten,
+  filterFlatten,
+  unflatten,
+  deleteFlatten,
+  mergeFlatten,
+  reject,
+} from "./ASTUtils";
+
+type Props = {
+  className?: string,
+  src: Object,
+};
+
+type State = {
+  src: Object,
+  flattenEmpty: Object,
+  flattenSrc: Object,
+  flattenType: Object,
+  flattenLocation: Object,
+  astOption: {
+    location: boolean,
+    empty: boolean,
+    type: boolean,
+  },
+};
+
+const OPTION_ORDER = ["location", "empty", "type"];
+
+export default class ASTPanel extends React.Component<Props, State> {
+  state = {
+    src: {},
+    flattenEmpty: {},
+    flattenSrc: {},
+    flattenType: {},
+    flattenLocation: {},
+    astOption: {
+      autofocus: true,
+      location: true,
+      empty: true,
+      type: true,
+    },
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    if (nextProps.src && nextProps.src !== prevState.src) {
+      let flattenSrc = flatten(nextProps.src);
+      flattenSrc = reject(flattenSrc, "_");
+      const src = unflatten(flattenSrc);
+
+      return {
+        src: src,
```

Comment:
When the input is changed (`props.src`) the options are ignored. So I disabled `location` but as soon as I change the input the locations are back. I have to enable and then disable `locations` again to remove locations.